strawberryfields/engine.py

```diff
@@ -182,58 +285,34 @@ def _run_program_locally(self, prog, **kwargs):
                 try:
                     temp = cmd.op.decompose(cmd.reg)
                     # run the decomposition
-                    applied_cmds = self._run_program_locally(temp, **kwargs)
+                    applied_cmds = self._run_program(temp, **kwargs)
                     applied.extend(applied_cmds)
                 except NotImplementedError as err:
                     # simplify the error message by suppressing the previous exception
                     raise err from None
         return applied
 
-    def run(self, program, return_state=True, modes=None, compile=True, **kwargs):
-        """Execute the given program by sending it to the backend.
+    def run(self, program, *, compile=True, modes=None, state_options={}, **kwargs):
+        """Execute the given programs by sending them to the backend.
 
-        * The backend state is updated.
-        * The executed Program(s) are appended to self.run_progs.
+        Extends :meth:`BaseEngine._run`.
 
         Args:
-            program (Program, Sequence[Program]): quantum circuit(s) to run
-            return_state (bool): If True, returns the state of the circuit after the
-                circuit has been run like :meth:`return_state` was called.
-            modes (Sequence[int]): Modes to be returned in the state object. If None, returns all modes.
-            compile (bool): If True, compile the Program instances before sending them to the backend.
-        """
-        if not isinstance(program, Sequence):
-            program = [program]
-
-        # TODO unsuccessful run due to exceptions should ideally have no effect on the backend state (state checkpoints?)
-        try:
-            prev = None  # previous program segment
-            for p in program:
-                if self.run_progs:
-                    prev = self.run_progs[-1]
-                else:
-                    # initialize the backend TODO where should this happen? c.f. the fixtures in conftest.py
-                    self.backend.begin_circuit(num_subsystems=p.init_num_subsystems, **self.kwargs)
+            modes (None, Sequence[int]): Modes to be returned in the ``Result.state`` :class:`.BaseState` object.
```

Comment:
Seems to be missing docstrings for other args?

Reply:
It's on purpose, there's a link to the parent class method that `LocalEngine.run()` extends. But I can also copy the docs for the parent method args here if you think it's clearer.

Reply:
Might be clearer to copy over docs related to `Args`, since people who are directly reading the source code won't see descriptions of all the args

Reply:
Ok, I'll copy them over.